subworkflows/local/differential/main.nf

```diff
@@ -0,0 +1,173 @@
+//
+// Perform differential analysis
+//
+include { PROPR_PROPD as PROPD } from "../../../modules/local/propr/propd/main.nf"
+include { DESEQ2_DIFFERENTIAL } from '../../../modules/nf-core/deseq2/differential/main'
+include { DESEQ2_DIFFERENTIAL as DESEQ2_NORM } from "../../../modules/nf-core/deseq2/differential/main"
+include { LIMMA_DIFFERENTIAL } from '../../../modules/nf-core/limma/differential/main'
+include { FILTER_DIFFTABLE as FILTER_DIFFTABLE_LIMMA } from '../../../modules/local/filter_difftable'
+include { FILTER_DIFFTABLE as FILTER_DIFFTABLE_DESEQ2 } from '../../../modules/local/filter_difftable'
+
+workflow DIFFERENTIAL {
+    take:
+    ch_counts             // [ meta_exp, counts ] with meta keys: method, args_diff
+    ch_samplesheet        // [ meta_exp, samplesheet ]
+    ch_contrasts          // [ meta_contrast, contrast_variable, reference, target ]
+
+    main:
+
+    // initialize empty results channels
+    ch_results_pairwise          = Channel.empty()
+    ch_results_pairwise_filtered = Channel.empty()
+    ch_results_genewise          = Channel.empty()
+    ch_results_genewise_filtered = Channel.empty()
+    ch_adjacency                 = Channel.empty()
+
+    // branch tools to select the correct differential analysis method
+    ch_counts
+        .branch {
+            propd:  it[0]["method"] == "propd"
+            deseq2: it[0]["method"] == "deseq2"
+            limma:  it[0]["method"] == "limma"
+        }
+        .set { ch_counts }
```

Comment:
I don't see your problem. The meta is existing in every nf-core module and subworkflow and from what I see it is also not uncommon to use the meta information for downstream analysis decisions. Take for example the single_end flag in RNAseq, ChIPseq, ATACseq,... Both modules (e.g. [trimgalore](https://github.com/nf-core/chipseq/blob/2.1.0/modules/nf-core/trimgalore/main.nf) ) and [subworkflows](https://github.com/nf-core/chipseq/blob/76e2382b6d443db4dc2396e6831d1243256d80b0/subworkflows/local/bam_filter_bamtools.nf#L37](https://github.com/nf-core/chipseq/blob/76e2382b6d443db4dc2396e6831d1243256d80b0/subworkflows/nf-core/fastq_fastqc_umitools_trimgalore/main.nf#L60) ) are using it, and thus forcing the flag to exist. This is even also used for [branching](https://github.com/nf-core/chipseq/blob/76e2382b6d443db4dc2396e6831d1243256d80b0/subworkflows/local/bam_filter_bamtools.nf#L37).

Reply:
The single-end thing is a special case, we try and discourage the propagation of assumptions about fields in the meta.

The branching you point at is necessitated by the need to direct outputs form the first processes to the right place. It's no engineered in to the inputs. If I were writing that subworkflow without that first process I'd be saying the same thing- the single end status would be a straighforward subworkflow input.

Reply:
The point here is that the tool status has nothing to do with the count file. You're forcing people to build it into the meta to suit the use case, but it would just make life harder for other users of the subworkflow.

Reply:
https://nf-co.re/docs/guidelines/components/modules#types-of-meta-fields